packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -158,6 +158,19 @@ void expectSync(dynamic actual, dynamic matcher, {
   test_package.expect(actual, matcher, reason: reason);
 }
 
```

Comment:
Mention `expectLater` in `expect`'s documentation.
Have `expect` assert that it didn't get a Future back, and if it did, fail with a message saying that you either have a bug or you should consider expectLater.

Reply:
`expect` can't get a future back, because it calls `test_package.expect()`, which returns void.

Reply:
Ugh, that's quite unfortunate.